platform/windows/context_gl_windows.cpp

```diff
@@ -63,16 +65,72 @@ int ContextGL_Windows::get_window_height() {
 	return OS::get_singleton()->get_video_mode().height;
 }
 
+bool ContextGL_Windows::should_vsync_via_compositor() {
+
+	if (OS::get_singleton()->is_window_fullscreen()) {
+		return false;
+	}
+
+	// Note: All Windows versions supported by Godot (Vista and later)
+	// have a compositor.  It can be disabled on earlier Windows versions.
+	BOOL dwm_enabled;
+
+	if (SUCCEEDED(DwmIsCompositionEnabled(&dwm_enabled))) {
+		return dwm_enabled;
+	}
```

Comment:
Do we need to call this function in every frame?

On Windows 8+ it's always true, and for Windows 7 it might be better to call it once at the start and then add `WM_DWMCOMPOSITIONCHANGED` message handler. See [DwmIsCompositionEnabled docs](https://docs.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmiscompositionenabled) for more info.

Reply:
This is what glfw and Chromium do.  `DwmIsCompositionEnabled()` is being called only when vsync is enabled so the thread will block when it calls either `Dwmflush()` or `SwapBuffers()`.  Because of this, I don't know if it is worth the trouble to listen for WM_DWMCOMPOSITIONCHANGED.